src/core/thread/csl_tx_scheduler.cpp

```diff
@@ -97,7 +98,7 @@ void CslTxScheduler::Clear(void)
 {
     for (Child &child : Get<ChildTable>().Iterate(Child::kInStateAnyExceptInvalid))
     {
-        child.SetCslTxAttempts(0);
+        child.ResetIndirectTxAttempts();
```

Comment:
The reason we separate TX attempts for CSL and indirect transmission is that, when CSL Tx fails for max attempts, we hope the message could still be successfully delivered by indirect transmission. If we use the same counter now, will the frame be discarded and cannot fetched by data poll any more after CSL Tx fails for max attempts?

Reply:
Yes. I think for a SSED, it is reasonable for it to use a very big data poll period(so as the save power) or even turn off data poll until it decides no packets are received during CSL timeout period and consider CSL out of sync. So I think it would be more practical to count CSL tx attempts as indirect sending attempts and just drop the packet if all the attempts fails.

For default setting, this attempt is 4. But in some products, this value could be something bigger like 16 which makes the waiting time way too long.

Reply:
Ok. Make sense to me.

Reply:
Thread 1.2 states the following:
```
Retries are handled using the mechanism defined for CSL in [IEEE 802.15.4-2015]. After
the maximum number of failed retransmissions, defined by macMaxFrameRetries, the Child
MUST be considered CSL de-synchronized.
```
and:
```
When a SSED Child is considered de-synchronized by the Parent, all packets to be
forwarded to the Child SHALL be sent by indirect transmission.
```
If we hit the max retries, then we will have to deal with incoming messages that may be queued for a long time waiting for a data poll.

Reply:
Thank you Jonathan for pointing this out. I think the rational behind the spec is, if CSL transmission reaches `macMaxFrameRetries`(set to 4 by default), it is big enough to consider this SSED out of sync. Thus there are no point of dropping this frame and trying sending another packet to this child. In this case, the parent can do nothing but just queue the packet and wait for data poll from this child.

This change violates the spec, I will change this part back to the original implementation.

Reply:
I have changed back to use separate counter for CSL tx attempts and indirect sending attempts. Packet will only be dropped when indirect transmission counter reaches max.

Also fix one issue that CSL tx attempts counter will never be reset once it reaches max(only got reset with a successful CSL transmission). This will stuck CSL transmission even this child regain sync later with this router. PR description also updated(point 2). Please help take a look. Thank you!